src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx

```diff
@@ -168,7 +174,13 @@ export class DataSourceSelector extends React.Component<
         return;
       }
 
-      const defaultDataSource = (await getDefaultDataSourceId(this.props.uiSettings)) ?? null;
+      const currentWorkspaceId = getWorkspaces().currentWorkspaceId$.getValue();
+      const scope: UiSettingScope = !!currentWorkspaceId
+        ? UiSettingScope.WORKSPACE
+        : UiSettingScope.GLOBAL;
+
+      const defaultDataSource =
+        (await getDefaultDataSourceId(this.props.uiSettings, scope)) ?? null;
```

Comment:
Nit: Any reason we pass scope from outside in `src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx`, but we get from workspace in current component?